src/DynamoCore/Models/NodeModel.cs

```diff
@@ -1612,19 +1611,19 @@ internal void RequestValueUpdateAsync(IScheduler scheduler, EngineController eng
                 VariableName = variableName
             });
 
-            task.Completed += OnNodeValueQueried;
+            task.Completed += QueryMirrorDataAsyncTaskCompleted;
             scheduler.ScheduleForExecution(task);
         }
 
-        private void OnNodeValueQueried(AsyncTask asyncTask)
+        private void QueryMirrorDataAsyncTaskCompleted(AsyncTask asyncTask)
         {
-            lock (cachedMirrorDataMutex)
+            asyncTask.Completed -= QueryMirrorDataAsyncTaskCompleted;
+
+            var task = asyncTask as QueryMirrorDataAsyncTask;
+            if (task != null)
             {
-                var task = asyncTask as QueryMirrorDataAsyncTask;
-                cachedMirrorData = task.MirrorData;
+                this.CachedValue = task.MirrorData;
             }
```

Comment:
I guess there's no need to un-register for the event, but good thing to do. For incorrect task type, instead of silently handle it, it'd be more appropriate to throw an exception wouldn't it?